Podcast/SearchDiscoverViewController.swift

```diff
@@ -267,23 +267,15 @@ class SearchViewController: ViewController, UISearchControllerDelegate, UITableV
     }
 
     @objc func searchAfterDelay() {
-        for data in self.tableViewData {
-            data.fetchData(query: lastSearchText)
-        }
+        tableViewData.fetchData(query: lastSearchText)
     }
-
-    // MARK: - Tab Bar Delegate
-
-    func selectedTabDidChange(toNewIndex newIndex: Int) {
-        addSearchITunesHeader()
-        searchResultsTableView.dataSource = tableViewData[newIndex]
-        searchResultsTableView.delegate = tableViewData[newIndex]
-        if let data = searchResultsTableView.dataSource as? SearchDataSourceDelegate, data.completingNewSearch {
-            searchResultsTableView.startLoadingAnimation()
-        } else {
-            searchResultsTableView.stopLoadingAnimation()
-        }
-        searchResultsTableView.reloadData()
+    
+    func searchBarCancelButtonClicked(_ searchBar: UISearchBar) {
+        discoverContainerView.isHidden = false
+    }
+    
+    func searchBarTextDidBeginEditing(_ searchBar: UISearchBar) {
```

Comment:
didBeginEditingText(on searchBar: UISearchBar)

Reply:
this too